prow/cmd/admission/dev.yaml

```diff
@@ -0,0 +1,90 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: prow-admission
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prow-admission
+  template:
+    metadata:
+      labels:
+        app: prow-admission
+    spec:
+      serviceAccount: prow-admission
+      containers:
+      - name: admission
+        image: gcr.io/k8s-testimages/admission:latest  # Note: not gcr.io/k8s-prow for dev
```

Comment:
Why do we need `dev.yaml` for this?

Reply:
This is useful do deploy the webhook during deployment. I will follow up with a production after this PR gets merged.